framework/sysdeps/windows/cpu_affinity.cpp

```diff
@@ -51,12 +59,20 @@ bool pin_to_logical_processor(LogicalProcessor n, const char *thread_name)
     if (n == LogicalProcessor(-1))
         return true;            // don't change affinity
 
+    HANDLE hThread = GetCurrentThread();
+
+#ifdef ENABLE_WINE_WORKAROUNDS
+    // Work around WINE not implementing SetThreadIdealProcessorEx
+    static const bool is_wine = running_in_wine();
+    if (is_wine && SetThreadIdealProcessor(hThread, DWORD(n)) == 0)
```

Comment:
This is probably causing the CI failures when run on a 2-core hyperthreaded CI (looks like AMD CPU).